### DISCUSSION TITLE
Can't build for arm with latest code

### DISCUSSION THREAD

**Discussion by benjaminplong** [ERGO-Code/HiGHS#1722] [General]
I updated my Highs src code to the latest master today and I am no longer able to build for arm. I am attempting the following using CMake 3.15.7 and Ubuntu 18.04: 
`
sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu binutils-aarch64-linux-gnu

mkdir build && cd build
cmake \
    -DCMAKE_C_COMPILER=aarch64-linux-gnu-gcc \
    -DCMAKE_CXX_COMPILER=aarch64-linux-gnu-g++ \
    -DCMAKE_SYSTEM_NAME=Linux \
    -DCMAKE_SYSTEM_PROCESSOR=aarch64 \
    -DFAST_BUILD=OFF \
    -DZLIB=OFF \
    ..
`

I am encountering:
-- The CXX compiler identification is GNU 7.5.0
-- The C compiler identification is GNU 7.5.0
-- Check for working CXX compiler: /usr/bin/aarch64-linux-gnu-g++
-- Check for working CXX compiler: /usr/bin/aarch64-linux-gnu-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/aarch64-linux-gnu-gcc
-- Check for working C compiler: /usr/bin/aarch64-linux-gnu-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- HIGHS version: 1.7.0
-- Build C++ library: ON
-- Build Fortran: OFF
-- Build CSharp: OFF
-- Build Python: OFF
-- Build all tests: OFF
-- ZLIB: OFF
-- Build type: Release
-- Looking for C++ include sys/types.h
-- Looking for C++ include sys/types.h - found
-- Looking for C++ include stdint.h
-- Looking for C++ include stdint.h - found
-- Looking for C++ include stddef.h
-- Looking for C++ include stddef.h - found
-- Check size of long
-- Check size of long - done
-- Found long size: 8
-- Check size of long long
-- Check size of long long - done
-- Found long long size: 8
-- Check size of int64_t
-- Check size of int64_t - done
-- Found int64_t size: 8
-- Check size of unsigned long
-- Check size of unsigned long - done
-- Found unsigned long size: 8
-- Check size of unsigned long long
-- Check size of unsigned long long - done
-- Found unsigned long long size: 8
-- Check size of uint64_t
-- Check size of uint64_t - done
-- Found uint64_t size: 8
-- Check size of int *
-- Check size of int * - done
-- Found int * size: 8
-- Performing Test HIGHS_HAVE_MM_PAUSE
-- Performing Test HIGHS_HAVE_MM_PAUSE - Failed
-- Performing Test HIGHS_HAVE_BUILTIN_CLZ
-- Performing Test HIGHS_HAVE_BUILTIN_CLZ - Success
-- Performing Test flag_supported
-- Performing Test flag_supported - Success
-- Performing Test flag_supported
-- Performing Test flag_supported - Success
-- Performing Test flag_supported
-- Performing Test flag_supported - Success
-- Performing Test flag_supported
-- Performing Test flag_supported - Success
-- Performing Test flag_supported
-- Performing Test flag_supported - Success
FLAG_MPOPCNT_SUPPORTED is not available on this architecture
-- Looking for a Fortran compiler
-- Looking for a Fortran compiler - NOTFOUND
-- Looking for a CSharp compiler
-- Looking for a CSharp compiler - NOTFOUND
-- Performing Test NO_OMIT_FRAME_POINTER_FLAG_SUPPORTED
-- Performing Test NO_OMIT_FRAME_POINTER_FLAG_SUPPORTED - Success
-- Git hash: 13363c9f1
-- Compilation date: 2024-04-09
-- No CSharp support
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB - Success
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB - Success
CMake Error at check/CMakeLists.txt:100 (cmake_path):
  Unknown CMake command "cmake_path".

-- Configuring incomplete, errors occurred!
See also "/home/blg/HiGHS/build/CMakeFiles/CMakeOutput.log".
See also "/home/blg/HiGHS/build/CMakeFiles/CMakeError.log".

[CMakeError.log](https://github.com/ERGO-Code/HiGHS/files/14922520/CMakeError.log)
[CMakeOutput.log](https://github.com/ERGO-Code/HiGHS/files/14922521/CMakeOutput.log)

Can someone help me make sense of the output? Building for x86_64 on this machine works fine.

**Comment by odow** [comment]
Use a newer version of CMake (3.20 or newer): https://cmake.org/cmake/help/latest/command/cmake_path.html

**Reply by benjaminplong** [reply]
I can try that but CMakeLists says min version is 3.15 and standard compilation works with 3.15.7

**Reply by odow** [reply]
> CMakeLists says min version is 3.15

@galabovaa lies 😆 

<img width="487" alt="image" src="https://github.com/ERGO-Code/HiGHS/assets/8177701/143ae55f-c3a0-4ed4-8f80-4152c6c44ff2">

**Reply by benjaminplong** [reply]
can confirm cmake 3.20 works fine. Thank you.
